sympy/printing/printer.py

```diff
@@ -55,135 +61,111 @@
+    from sympy import Symbol
+    from sympy.printing.latex import LatexPrinter, print_latex
+    from sympy.core.function import UndefinedFunction, Function
 
-    Its job is to provide infrastructure for implementing new printers easily.
 
-    Basically, if you want to implement a printer, all you have to do is:
+    class MyLatexPrinter(LatexPrinter):
+        \"\"\"Print derivative of a function of symbols in a shorter form.
+        \"\"\"
+        def _print_Derivative(self, expr):
+            function, *vars = expr.args
+            if not isinstance(type(function), UndefinedFunction) or \\
+               not all(isinstance(i, Symbol) for i in vars):
+                return super()._print_Derivative(expr)
 
-    1. Subclass Printer.
+            return "{}_{{{}}}".format(
+                self._print(Symbol(function.func.__name__)),
+                            ''.join(self._print(i) for i in vars))
 
-    2. Define Printer.printmethod in your subclass.
-       If a object has a method with that name, this method will be used
-       for printing.
 
-    3. In your subclass, define ``_print_<CLASS>`` methods
+    def print_my_latex(expr):
+        \"\"\" Most of the printers define their own wrappers for print().
+        These wrappers usually takes printer settings. Our printer does not have
+        any settings.
+        \"\"\"
+        print(MyLatexPrinter().doprint(expr))
 
-       For each class you want to provide printing to, define an appropriate
-       method how to do it. For example if you want a class FOO to be printed in
-       its own way, define _print_FOO::
 
-           def _print_FOO(self, e):
-               ...
+    y = Symbol("y")
+    x = Symbol("x")
+    f = Function("f")
+    expr = f(x, y).diff(x, y)
 
-       this should return how FOO instance e is printed
+    # Print the expression using the normal latex printer and our custom
+    # printer.
+    print_latex(expr)
+    print_my_latex(expr)
 
-       Also, if ``BAR`` is a subclass of ``FOO``, ``_print_FOO(bar)`` will
-       be called for instance of ``BAR``, if no ``_print_BAR`` is provided.
-       Thus, usually, we don't need to provide printing routines for every
-       class we want to support -- only generic routine has to be provided
-       for a set of classes.
+The output of the code above is::
 
-       A good example for this are functions - for example ``PrettyPrinter``
-       only defines ``_print_Function``, and there is no ``_print_sin``,
-       ``_print_tan``, etc...
+    \\frac{\\partial^{2}}{\\partial x\\partial y}  f{\\left (x,y \\right )}
+    f_{xy}
 
-       On the other hand, a good printer will probably have to define
-       separate routines for ``Symbol``, ``Atom``, ``Number``, ``Integral``,
-       ``Limit``, etc...
+Example of Custom Printing Method
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-    4. If convenient, override ``self.emptyPrinter``
+In this example below, we want to print modulo operator in a different way in
```

Comment:
->
In the example below, the latex printing of the modulo operator is modified. This
is done by overriding the method ``_latex`` of ``Mod``.